lib/gui/lvgl/Kconfig.graphical

```diff
@@ -22,6 +22,26 @@ config LVGL_VER_RES_MAX
 	help
 	  Vertical screen resolution in pixels
 
+choice LVGL_DISPLAY_ROTATION
```

Comment:
Can you please explain why these new options are needed? And how is it related to "lib: gui: lvgl: get display rotation from driver for correct drawing ……X and Y"?

Reply:
For example, we have a 320x240 display
We have specified this width=320 and height=240, All display drivers have default rotation on initialization,
After we got binded up with driver, since we have a rotation function, we need to send this to the driver before initializing the LVGL layer.
The user can choose display rotation or default position.

And it would be very cool to add functionality to rotate the display in LVGL layer while the program is running.
I am using display and accelerometer in my project.
The rotation of the display depends on the position of the device, or can be turned off and used by default rotation 

For this feature, need to add a rotation API and a relocation buffer for drawing.
Because before that a buffer was located for drawing 320x240, but we need 240x320, 
https://github.com/zephyrproject-rtos/zephyr/blob/10c971a463b83eebdcf7b0d4edc8bb8df5b8a4f5/lib/gui/lvgl/lvgl.c#L126

Reply:
> For example, we have a 320x240 display
We have specified this width=320 and height=240, All display drivers have default rotation on initialization,
After we got binded up with driver, since we have a rotation function, we need to send this to the driver before initializing the LVGL layer.

Initial rotation can be set for some drivers: https://github.com/zephyrproject-rtos/zephyr/blob/af79664da3debec8d168e01dac6098dac88841cf/boards/shields/adafruit_2_8_tft_touch_v2/adafruit_2_8_tft_touch_v2.overlay#L21-L23

Reply:
Rotation display implemented only ILI9XXX display, in other displays, not

I make PR #40845 and add rotation feature to st7789v driver

see st7789v, st7735r
https://github.com/zephyrproject-rtos/zephyr/blob/af79664da3debec8d168e01dac6098dac88841cf/boards/shields/st7789v_generic/st7789v_waveshare_240x240.overlay#L12-L37

https://github.com/zephyrproject-rtos/zephyr/blob/af79664da3debec8d168e01dac6098dac88841cf/boards/shields/st7735r/st7735r_ada_160x128.overlay#L11-L34


I think it's a bad approach to define it in dts.
We need the ability to change orientation during work

Reply:
DT is used to describe hardware initial configuration, so it's a reasonable place to add this configuration. In some boards, display is mounted in a way that requires default rotation to be changed.

Runtime rotation is now possible using the display API, LVGL is not aware if rotation is changed, though. I guess an option could be to extend display API with notification capability about display rotation change, so that when an application calls `display_set_rotation`, LVGL gets notified and does necessary adjustments.

Reply:
Great! Sounds good
Only I think that If the user has a choice, define it in the DT and, if necessary, change the orientation during work, it would be great at all

Who will design and implement the API?
I can test as I am tightly working with display and LVGL in my project

Reply:
You can open a proposal, maybe present it to the API meeting (it happens weekly). You can also ask for feedback first on Discord, there is a display channel.

Reply:
Thanks @gmarull